sdk/dotnet/Pulumi/X/Automation/IWorkspace.cs

```diff
@@ -0,0 +1,198 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace Pulumi.X.Automation 
+{
+    /// <summary>
+    /// Workspace is the execution context containing a single Pulumi project, a program, and multiple stacks.
+    /// <para/>
+    /// Workspaces are used to manage the execution environment, providing various utilities such as plugin
+    /// installation, environment configuration ($PULUMI_HOME), and creation, deletion, and listing of Stacks.
+    /// </summary>
+    public interface IWorkspace 
```

Comment:
For .NET, we may want to model this as an abstract class (e.g. `public abstract class Workspace`) rather than as an interface to give us more flexibility to add members in the future.

Reply:
We could do that. Is there some functionality that is currently is the LocalWorkspace implementations that would belong more in an abstract Workspace class?

Reply:
There's also that new C# feature for default method implementations on interfaces :) don't know if that would be useful here though if you're just thinking about future dev

Reply:
> Is there some functionality that is currently in the LocalWorkspace implementations that would belong more in an abstract Workspace class?

It's possible. Might be easiest to start with implementing on `LocalWorkspace` and then see if there's anything general enough that could be provided as a default implementation on the base `Workspace`.